File: odin/arr.c
#include <stdio.h>
#include <stdlib.h>


int *do_job(int size) {
    int *buf = malloc(sizeof(int) * size);

    for(int i = 0; i < size; i++) {
        buf[i] = i + 1;
    }
    return buf;
}

int main() {

    int *buf = do_job(10000000);

    printf("%d\n", buf[9999999]);

    free(buf);

    return 0;
}

File: odin/arr.zig
const std = @import("std");


fn do_job(allocator: std.mem.Allocator, size: usize) ![]i32 {
    var buf = try allocator.alloc(i32, size);
    for (0..buf.len) |i| {
        buf[i] = @as(i32, @intCast(i + 1));
    }

    return buf;
}

pub fn main() !void {
    var allocator = std.heap.c_allocator;

    const buf = try do_job(allocator, 10000000);
    defer allocator.free(buf);


    std.debug.print("{d}\n", .{buf[9999999]});

}
